### ISSUE TITLE
Implement an Initial Question Extraction Function

### ISSUE THREAD

**Issue by grayJiaaoLi** [amosproj/amos2024ss08-cloud-native-llm#29]
## User story
1. As a developer
2. I need to create a basic tool that can identify questions from the extracted CNCF text
3. So that the Qs can be prepared for matching the As

## Acceptance criteria
* Identify sentences that can be formatted as questions

* Determine potential question structures based on the extracted data

* Examine the quality of extracted questions
    * Log the errors
    * Hold the broken texts to avoid disturbing

* Document the matching logic and limitations for further development
    * observed problems
    * extraction strategy
    * matching algorithm or mechanism

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by christianwielenberg** [comment]
Why is this not closed?
